Branch-SDK/src/main/java/io/branch/coroutines/InstallReferrers.kt

```diff
@@ -230,6 +231,97 @@ suspend fun getXiaomiGetAppsReferrerDetails(context: Context): InstallReferrerRe
     }
 }
 
+suspend fun getMetaInstallReferrerDetails(context: Context): InstallReferrerResult? = withContext(Dispatchers.Default) {
+    try {
+        val fbAppID = PrefHelper.fbAppId_
+
+        if (fbAppID.isNullOrEmpty()) {
+            BranchLogger.d("No Facebook App ID provided. Can't check for Meta Install Referrer")
+            null
+        } else {
+            queryMetaInstallReferrer(context, fbAppID)
+        }
+    } catch (exception: Exception) {
+        BranchLogger.e("Exception in getMetaInstallReferrerDetails: $exception")
+        null
+    }
+}
+private fun queryMetaInstallReferrer(context: Context, fbAppId: String): InstallReferrerResult? {
+    val facebookProvider = "content://com.facebook.katana.provider.InstallReferrerProvider/$fbAppId"
+    val instagramProvider = "content://com.instagram.contentprovider.InstallReferrerProvider/$fbAppId"
+
+    val facebookResult = queryProvider(context, facebookProvider)
+    val instagramResult = queryProvider(context, instagramProvider)
+
+    // Check both Facebook and Instagram for install referrers and return the latest one
+    return if (facebookResult != null && instagramResult != null) {
```

Comment:
While this is one return statement, it's a bit (for me) to read. Stylistically I would suggest assigning a result variable and returning just that.

See the section on coalescing returns https://android-developers.googleblog.com/2023/11/the-secret-to-androids-improved-memory-latest-android-runtime-update.html